src/session.rs

```diff
@@ -164,12 +164,20 @@ pub trait NetSession: NetStream + SplitIo {
     fn disconnect(self) -> io::Result<()>;
 }
 
+#[derive(Clone, Debug, Display, Error)]
+#[display("handshake has failed due to {0}")]
```

Comment:
```suggestion
#[display("handshake failed: {0}")]
```
I think this will read better generally

Reply:
I am trying to avoid `:` in error messages, since the actual report otherwise will look like

`Error: connection failed: handshake failed: remote key is invalid`.

This, I make them look like

`Error: connection was not established because handshake has failed due to remote key being invalid`.

Reply:
Ok I see, the problem is you don't know what's in `{0}`, it could be "failed to open stream" or something and then you have `handshake has failed due to failed to open stream` which isn't correct english.. So you just have to make sure the inner errors follow some kind of format then.

Reply:
Yes, true. I constantly improve that messages such that they can work together.